src/nvapi/nvapi_d3d12_device.h

```diff
@@ -5,46 +5,36 @@
 #include "../util/com_pointer.h"
 
 namespace dxvk {
-
     class NvapiD3d12Device {
 
-        struct CommandListExtWithVersion {
-            ID3D12GraphicsCommandListExt1* CommandListExt;
-            uint32_t InterfaceVersion;
-        };
-
       public:
-        static std::optional<LUID> GetLuid(IUnknown* unknown);
+        static void Reset();
+        [[nodiscard]] static NvapiD3d12Device* GetOrCreate(ID3D12Device* device);
+        [[nodiscard]] static uint32_t FindCubinSmem(NVDX_ObjectHandle);
+
+        explicit NvapiD3d12Device(ID3D12DeviceExt* vkd3dDevice);
 
+        static std::optional<LUID> GetLuid(IUnknown* unknown);
         static bool CreateGraphicsPipelineState(ID3D12Device* device, const D3D12_GRAPHICS_PIPELINE_STATE_DESC* pipelineStateDescription, NvU32 numberOfExtensions, const NVAPI_D3D12_PSO_EXTENSION_DESC** extensions, ID3D12PipelineState** pipelineState);
         static bool SetDepthBoundsTestValues(ID3D12GraphicsCommandList* commandList, float minDepth, float maxDepth);
```

Comment:
I'm not yet sure if this should stay here. For one, none of these really access anything NVAPI-specific… at least not yet. Ann top of that:

* `GetLuid` is used only in `NvAPI_D3D12_GetGraphicsCapabilities`, and only because the input is typed as `IUnknown` :weary:;
* `CreateGraphicsPipelineState` just checks for one known extension and forwards to core D3D12 function otherwise.
* `SetDepthBoundsTestValues` upgrades the interface to GCL1 then calls core D3D12 function… but even if you want to avoid inlining this, it should probably live in `NvapiD3d12GraphicsCommandList` instead.

Reply:
I don't mind them here, but also ok to move those, so moved down them.

Reply:
PS: will also inline the other two.